editor/editor_scale.cpp

```diff
@@ -29,11 +29,16 @@
 /**************************************************************************/
 
 #include "editor_scale.h"
+#include "core/math/math_funcs.h"
 
 float EditorScale::_scale = 1.0f;
 
 void EditorScale::set_scale(float p_scale) {
-	_scale = p_scale;
+	if (p_scale <= 0 || Math::is_nan(p_scale)) {
```

Comment:
```suggestion
	if (!(p_scale > 0)) {
```
Also has the same effect

Reply:
@AThousandShips  I use the Math::is_nan because I think to detect nan, one must use one of these functions or macros.  A NaN is generated when the result of a floating-point operation can't be represented in IEEE-754 floating-point format for the specified type (from https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/isnan-isnan-isnanf?view=msvc-170). Or maybe I missed something here, thank you for your feedback!

Reply:
As I said this detects nan, and is regularly used in the engine for this

I am well aware of NaN and how it works

Reply:
Also it might actually be better to do:
```suggestion
	if (p_scale <= 0 || !Math::is_finite(p_scale)) {
```
As `inf` might also be breaking things, so either this or my suggestion above would probably be best

Reply:
Just make sure to swap the branches and remove the negation in the condition.

Reply:
Thanks !